pkg/sql/set_zone_config.go

```diff
@@ -185,9 +184,6 @@ func (p *planner) SetZoneConfig(ctx context.Context, n *tree.SetZoneConfig) (pla
 	if err := checkPrivilegeForSetZoneConfig(ctx, p, n.ZoneSpecifier); err != nil {
 		return nil, err
 	}
-	if !p.ExecCfg().Codec.ForSystemTenant() {
```

Comment:
We need to remove these `if [..]Codec.ForSystemTenant()` checks below, for `RemoveIndexZoneConfigs`. I think we'd benefit from a general audit of all these cases. Maybe lets just try and port over the system tenant zone config logic tests to the tenants right now, almost all of them should apply (save for the distinguished ranges, and for replication reports). That should shake out all these tests.